packages/mux-uploader/src/index.ts

```diff
@@ -0,0 +1,419 @@
+</div>
+<div class="radial-type">
+  <svg
+    width="120"
+    height="120">
+    <!-- To prevent overflow of the SVG wrapper, radius must be  (svgWidth / 2) - (circleStrokeWidth * 2)
+      or use overflow: visible on the svg.-->
+    <circle
+      r="52"
+      cx="60"
+      cy="60"
+    />
+  <svg>
+</div>
+
+<slot name="dropzone">
+  <mux-uploader-drop></mux-uploader-drop>
+</slot>
+`;
+
+// Note: Use "bar" for now since the CSS for radial is WIP. (TD).
+const TYPES = {
+  BAR: 'bar',
+  RADIAL: 'radial',
+};
+
+class MuxUploaderElement extends HTMLElement {
+  hiddenFileInput: HTMLInputElement | null | undefined;
+  filePickerButton: HTMLButtonElement | null | undefined;
+  svgCircle: SVGCircleElement | null | undefined;
+  progressBar: HTMLElement | null | undefined;
+  uploadPercentage: HTMLElement | null | undefined;
+  statusMessage: HTMLElement | null | undefined;
+  retryMessage: HTMLElement | null | undefined;
+  _dropHandler: Function;
+
+  constructor() {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+    const uploaderHtml = template.content.cloneNode(true);
+    shadow.appendChild(uploaderHtml);
+
+    this.hiddenFileInput = this.shadowRoot?.querySelector('input[type="file"]');
+    this.filePickerButton = this.shadowRoot?.querySelector('button');
+    this.svgCircle = this.shadowRoot?.querySelector('circle');
+    this.progressBar = this.shadowRoot?.getElementById('progress-bar');
+    this.uploadPercentage = this.shadowRoot?.getElementById('upload-status');
+    this.statusMessage = this.shadowRoot?.getElementById('status-message');
+    this.retryMessage = this.shadowRoot?.getElementById('retry-message');
+
+    this._dropHandler = this.handleUpload.bind(this);
+  }
+
+  connectedCallback() {
+    this.setDefaultType();
+    this.setupFilePickerButton();
+    this.setupRetry();
+    this.setupDropHandler();
+
+    this.setAttribute('role', 'progressbar');
+    this.setAttribute('aria-label', 'progress bar');
+    this.setAttribute('aria-live', 'polite');
+
+    // TO-DO: Might want to standardize if we prefer to have users disable or enable things. (TD).
+    // Edge case: User wants to use the uploader without drag.
+    // Because we slot a default mux-uploader-drop if they don't slot one,
+    // in order to disable drop, the user has to pass disableDrop to mux-uploader and mux-uploader-drop
+    // must apply it in order to disable the drop. It currently requires you to explicit pass "true".
+    if (this.getAttribute('disableDrop')) {
+      const muxUploaderDrop = this.shadowRoot?.querySelector('mux-uploader-drop');
+      muxUploaderDrop?.setAttribute('disableDrop', '');
+    }
+  }
+
+  disconnectedCallback() {
+    //@ts-ignore
+    this.removeEventListener('mux-drop', this._dropHandler, false);
+  }
+
+  get radius() {
+    return Number(this.svgCircle?.getAttribute('r'));
+  }
+
+  get circumference() {
+    return this.radius * 2 * Math.PI;
+  }
```

Comment:
helper functions like these better are defined out of the class with the class instance as an argument so they're not exposed on the public API. 
e.g. `const getRadius = (el) => Number(el.svgCircle?.getAttribute('r'));`